### PR TITLE
Add examples for using a pact broker.

### PR DESCRIPTION
**Pull request by Alexander-Miller**

### CODE REVIEW

File: pact/http/consumer-one/build.gradle.kts [hunk 1]
```diff
@@ -15,3 +16,14 @@ dependencies {
     testImplementation("org.springframework.boot:spring-boot-starter-test")
     testImplementation("au.com.dius.pact.consumer:junit5")
 }
+
+project.version = "1.0.0"
```
Comment (slu-it) [inline pact/http/consumer-one/build.gradle.kts]: Could this be `0.1.0-SNAPSHOT`?

File: pact/http/consumer-one/build.gradle.kts [hunk 2]
```diff
@@ -15,3 +16,14 @@ dependencies {
...
+
+pact {
+    publish {
+        pactBrokerUrl = "http://localhost:9292"
+        consumerVersion = project.version
```
Comment (slu-it) [inline pact/http/consumer-one/build.gradle.kts]: Could this be set directly and the `project.version` left unset? Or are both needed for the plugin to work?

File: pact/http/consumer-two/build.gradle.kts [hunk 3]
```diff
@@ -15,3 +16,15 @@ dependencies {
...
+project.version = "1.5.0"
+
+pact {
+    publish {
+        pactBrokerUrl = "http://localhost:9292"
+        consumerVersion = project.version
+        pactBrokerUsername = "pact-user"
+        pactBrokerPassword = "123"
+    }
+}
```
Comment (slu-it) [inline pact/http/consumer-two/build.gradle.kts:27]: see comments / questions in `consumer-one`

File: pact/http/provider/build.gradle.kts [hunk 4]
```diff
@@ -17,3 +18,12 @@ dependencies {
     testImplementation("au.com.dius.pact.provider:junit5")
     testImplementation("io.mockk:mockk")
 }
+
+project.version = "3.2.1"
+
+tasks {
+    withType<Test> {
+        systemProperty("pact.verifier.publishResults", "true")
+        systemProperty("pact.provider.version", project.version)
```
Comment (slu-it) [inline pact/http/provider/build.gradle.kts]: Same questions as for the consumers: is `project.version` needed, or could it be set directly without issues?

A bit of my thoughts on why I would prefer it without the `project.version`, if possible: 
For a real microservice I would not set a project version. If a version is needed (e.g. if the Service JAR needs to be published before release), I usually just set it during release builds. Not sure how common this approach is though.

File: pact/http/provider/src/test/kotlin/provider/ContractTestWithPactBroker.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+// Verification results will be uploaded every time the test is run,
+// based on the value of the `pact.verifier.publishResults` property
+
+@TestInstance(PER_CLASS) // PACT needs test instance per class ... for some reason ...
+@Provider("provider") // load all contracts for the provider named "provider"
+@PactBroker( // configure the connection to the pact broker and which pacts to use
+    url = "http://localhost:9292",
+    authentication = PactBrokerAuth(
+        username = "pact-user",
+        password = "123"
```
Comment (slu-it) [inline pact/http/provider/src/test/kotlin/provider/ContractTestWithPactBroker.kt:47]: Not a change request, but a question for future optimization:
Did you find out if and how credentials could be externalized? e.g. System Properties?

File: build.gradle.kts [hunk 6]
```diff
@@ -7,6 +7,7 @@ plugins {
     id("org.springframework.boot") version "2.7.0" apply false
     id("io.spring.dependency-management") version "1.0.11.RELEASE" apply false
     id("org.springframework.cloud.contract") version "3.1.1" apply false
+    id("au.com.dius.pact") version "4.3.10" apply false
```
Comment (slu-it) [inline build.gradle.kts:10]: Currently, the project uses version `4.3.6` of the PACT libraries. Maybe we should update those to the same version?

**Review by slu-it — CHANGES_REQUESTED** [review]

File: pact/http/consumer-one/build.gradle.kts (hunk 1 above)
Comment (Alexander-Miller) [reply pact/http/consumer-one/build.gradle.kts]: Could be whatever. I just wanted round numbers for the broker frontend.

File: pact/http/consumer-one/build.gradle.kts (hunk 2 above)
Comment (Alexander-Miller) [reply pact/http/consumer-one/build.gradle.kts]: They can be set independently. Using `project.version` just ensures they always align.

File: pact/http/consumer-two/build.gradle.kts (hunk 3 above)
Comment (Alexander-Miller) [reply pact/http/consumer-two/build.gradle.kts:27]: done

File: pact/http/provider/build.gradle.kts (hunk 4 above)
Comment (Alexander-Miller) [reply pact/http/provider/build.gradle.kts]: I've inlined the versions now.

File: pact/http/provider/src/test/kotlin/provider/ContractTestWithPactBroker.kt (hunk 5 above)
Comment (Alexander-Miller) [reply pact/http/provider/src/test/kotlin/provider/ContractTestWithPactBroker.kt:47]: The free version of the broker (there's an enterprise tier at pactflow) only supports basic auth for read-ony/read-write access. For anything more sophisticated you'll have to pay. So externalization is doable via the usual means like environment variables etc.

File: build.gradle.kts (hunk 6 above)
Comment (Alexander-Miller) [reply build.gradle.kts:10]: done

**Review by slu-it — APPROVED** [review]

**Pull request merged**